File: app/src/main/java/com/example/ama/questapp/domain/repointerface/QuestPatternsRepository.java
package com.example.ama.questapp.domain.repointerface;

import com.example.ama.questapp.data.db.model.QuestPattern;
import java.util.List;

import io.reactivex.Single;

public interface QuestPatternsRepository {

    Single<List<QuestPattern>> getUnusedDailyQuestPatterns();

    Single<List<QuestPattern>> getRepeatableDailyQuestPatterns();

}

File: app/src/main/java/com/example/ama/questapp/domain/repointerface/QuestPatternsRepository.kt
package com.example.ama.questapp.domain.repointerface

import com.example.ama.questapp.data.db.model.QuestPattern
import io.reactivex.Single

interface QuestPatternsRepository {

    fun getUnusedDailyQuestPatterns(): Single<List<QuestPattern>>

    fun getRepeatableDailyQuestPatterns(): Single<List<QuestPattern>>
}